dashboard/app/models/user_professional_learning_course_enrollment.rb

```diff
@@ -16,21 +16,22 @@
 class UserProfessionalLearningCourseEnrollment < ActiveRecord::Base
   belongs_to :user
   belongs_to :professional_learning_course
-  has_many :user_enrollment_module_assignment
+  has_many :user_enrollment_module_assignment, dependent: :destroy
```

Comment:
Oh, the convention for a has many is to pluralize the name -- so it is usually `has_many :user_enrollment_module_assignments`. Maybe that's why you have to do things like `user_module_task_assignment.all.map`.